scripts/exglobal_prep_sfc.sh

```diff
@@ -0,0 +1,196 @@
+#                              from previous cycle.  Backup data if
+#                              program failure
+#  Output Files:
+#    $BLENDED_ICE_FILE - global 5-minute blended ice data (grib 1)
+#    $MODEL_SNOW_FILE  - snow analysis on model grid (grib 1)
+#
+# Condition codes:
+#  0       - normal termination
+#  non-0   - problem in the prep_sfc_ice_blend.sh or prep_sfc_snow.sh
+#            script and the backup ice or snow data is missing.
+#            fatal error.
+#
+# Attributes:
+#     Language:  RedHat Linux
+#     Machine:   NCEP WCOSS
+#
+#########################################################################
+
+##############################################################
+# Input data for emcsfc_ice_blend and emcsfc_snow2mdl programs
+##############################################################
+
+# ims snow cover and ice cover data (grib 1 or grib 2)
+export IMS_FILE="${COMINobsproc}/${RUN}.${cycle}.imssnow96.grib2"
+
+# global 5-minute ice concentration file (grib 2)
+export FIVE_MIN_ICE_FILE="${COMINobsproc}/${RUN}.${cycle}.seaice.5min.grib2"
+
+# landmask file for global 5-minute data (grib 2)
+export FIVE_MIN_ICE_MASK_FILE="${FIXgfs}/am/emcsfc_gland5min.grib2"
+
+# afwa snow depth data
+export AFWA_GLOBAL_FILE="${COMINobsproc}/${RUN}.${cycle}.snow.usaf.grib2"
+export AFWA_NH_FILE="${COMINobsproc}/${RUN}.${cycle}.NPR.SNWN.SP.S1200.MESH16.grb"
+export AFWA_SH_FILE="${COMINobsproc}/${RUN}.${cycle}.NPR.SNWS.SP.S1200.MESH16.grb"
+
+# the output ice blend data (grib)
+export BLENDED_ICE_FILE="${RUN}.${cycle}.seaice.5min.blend.grb"
+
+# the 6-hour old output ice blend data.
+export BLENDED_ICE_FILE_PREV="${COMINobsproc_PREV}/${RUN}.${gcycle}.seaice.5min.blend.grb"
+
+# the emcsfc_ice_blend executable
+export BLENDICEEXEC=${BLENDICEEXEC:-${EXECgfs}/emcsfc_ice_blend}
+
+# standard output file
+export pgmout=${pgmout:-OUTPUT}
+
+#-----------------------------------------------------------------------
+# call utility script to create global ice blend data.
+#-----------------------------------------------------------------------
+
+echo "Create blended ice data."
+"${USHgfs}/prep_sfc_ice_blend.sh"
+export err=$?
+
+#-----------------------------------------------------------------------
+# If there is a failure in the ice blend script, copy the 6-hr old file
+# to the current file.  The gfs/gdas can run with old ice data for
+# about a week.  Although not fatal, any errors must be
+# investigated.
+#
+# If there is a failure in the ice blend script AND the 6-hour old file
+# is not available as a backup, abort the script.  The global cycling
+# code can't run without an ice analysis.
+#-----------------------------------------------------------------------
+
+if [[ ${err} -ne 0 ]]; then
+  if [[ -s "${BLENDED_ICE_FILE_PREV}" ]]; then
+    echo "Copy old ice blend file to current directory"
+    cpfs "${BLENDED_ICE_FILE_PREV}" "${COMOUT_OBS}/${BLENDED_ICE_FILE}"
```

Comment:
Should probably set `err` back to 0.

```suggestion
    cpfs "${BLENDED_ICE_FILE_PREV}" "${COMOUT_OBS}/${BLENDED_ICE_FILE}"
    export err=0
```